app/models/manageiq/providers/vmware/infra_manager/vm_or_template_shared/disconnect.rb

```diff
@@ -4,14 +4,14 @@ module ManageIQ::Providers::Vmware::InfraManager::VmOrTemplateShared::Disconnect
   def disconnect_storage
     # If the VM was unregistered don't clear the storage because the disks
     # are still on the underlying datastore
-    super unless unregistered?
+    super unless vm_unregistered?
   end
 
-  def destroyed?
+  def vm_destroyed?
```

Comment:
omg, were we not calling `destroyed?` anywhere?   🤣 🤮

Reply:
Now we know, it was being called, just not by us... by rails 5.2 😆

Reply:
Yeah it was only added for "symmetry" with `unregistered?` :laughing: oh the irony

Reply:
> Yeah it was only added for "symmetry" with `unregistered?` 😆 oh the irony

This is amazing.  This could manifest as such weird errors all over the place and for "seemingly random vms".  Very ironic.